### ISSUE TITLE
Missing Headings for Cards

### ISSUE THREAD

**Issue by puja-iaccessible** [adobe/helix-block-collection#9]
**Actual Behaviour**
There are no headings on each card which makes it difficult to navigate for screen reader users. Currently there is only heading on the page (Another List).

**Expected Behaviour**
The absence of headings is not a WCAG violation. Having a heading on each card makes the cards more usable for screen reader users.

Please add a heading in the example output for the Card component for each card. This makes navigating through cards much easier for screen reader users. Adding it to sample components makes it much more likely that users of this component will also follow the pattern.

** Steps to Reproduce**
1. Go to the URL https://main--helix-block-collection--adobe.hlx.page/block-collection/cards
2. Use Accessibility Insights (Chrome/Edge Extension) to highlight the headings for the page
3. Notice that the cards do not have a heading and it is hard for screen reader users to tell when a new card begins.

**Applicable Guideline:**
This is more from a usability perspective and not a WCAG violation

**Screenshot**
![No Heading for Cards - only one heading on the page](https://user-images.githubusercontent.com/123964084/219414272-4244cc9a-9943-48e5-b781-3fac3e1dc6fc.png)
